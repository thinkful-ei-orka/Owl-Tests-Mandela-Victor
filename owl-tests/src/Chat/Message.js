import React from 'react';

export default function Message(props) {
     return (
          <li className='Chat-message'>
               <div className='Chat-message-header'>
                    <img src={props.avatar} alt='Participant Avatar'></img>
                    <h3>
                         {props.name}
                         <span>
                              {props.time}
                         </span>
                    </h3>
               </div>
               <p>
                    {props.message}
               </p>
          </li>
     )
}