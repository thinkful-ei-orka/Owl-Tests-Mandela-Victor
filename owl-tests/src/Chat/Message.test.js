import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Message from './Message';

describe('Message Component', () =>{
    let messageTest = [{
        avatar: 'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
        name: 'Koren Templeton',
        message: 'Hello World!'
    }]

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Message avatar={messageTest.avatar} name={messageTest.name} message={messageTest.message} />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<Message avatar={messageTest.avatar} name={messageTest.name} message={messageTest.message} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})