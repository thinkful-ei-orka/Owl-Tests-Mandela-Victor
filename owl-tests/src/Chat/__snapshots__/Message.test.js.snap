// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Component renders the UI as expected 1`] = `
<li
  className="Chat-message"
>
  <div
    className="Chat-message-header"
  >
    <img />
    <h3>
      <span />
    </h3>
  </div>
  <p />
</li>
`;
